tutorials/HDF5/Subset-HDF5-hyperspectral.Rmd

```diff
@@ -0,0 +1,241 @@
+***
+
+### Recommended Skills
+
+If you haven't already, we highly recommend you work through the 
+<a href="{{ site.baseurl }}/intro-hdf5-r-series" target="_blank"> *Introduction to Working with HDF5 Format in R* series</a>
+before moving on to this tutorial.
+
+</div>
+
+
+## Why subset your dataset? 
+There are many reasons why you may wish to subset you HDF5 file.
+Primarily, HDF5 files may contain a large amount of information
+that is not necessary for your purposes. By subsetting the file,
+you can reduce file size, thereby shrinking your storage needs,
+and shortening file transfer/download times. In this example, we
+will take a full HDF5 file of NEON hyperspectral reflectance data
+from the San Joaquin Experimental Range (SJER) that has a file size
+of ~`652 Mb` and make a new HDF5 file with a reduced spatial extent,
+and a reduced spectral resolution, yeilding a file of only ~`50.1 Mb`.
+This reduction in file size will make it easier and faster to share
+your data with others. We will then use this subsetted file in the
+<a href="{{ site.baseurl }}/intro-hsi-r-series" target="_blank"> *Introduction to Hyperspectral Remote Sensing Data* series</a>.
+
+If you find that downloading the full `652 Mb` file takes too much
+time or storage space, you will find a link to this subsetted file
+at the top of each lesson in the <a href="{{ site.baseurl }}/intro-hsi-r-series" target="_blank"> *Introduction to Hyperspectral Remote Sensing Data* series</a>.
+
+## Reading in the HDF5 file
+
+First, we will load the required library for this tutorial:
+
+```{r load-libraries}
+
+# Install rhdf5 package (only need to run if not already installed)
+# install.packages("BiocManager")
+# BiocManager::install("rhdf5")
+
+# Load required packages
+library(rhdf5)
+
+```
+
+Next, we define our working directory where we have saved the full HDF5 
+file of NEON hyperspectral reflectance data from the SJER site. Note,
+the filepath to the working directory will depend on your local environment.
+Then, we create a string (`f`) of the HDF5 filename and read its attributes.
+
+```{r read-in-file}
+# set working directory to ensure R can find the file we wish to import and where
+# we want to save our files. Be sure to move the download into your working directory!
+wd="~/Desktop/Hyperspectral_Tutorial/" #This will depend on your local environment
+setwd(wd)
+
+# Read in H5 file
+f <- paste0(wd,"NEON_D17_SJER_DP3_257000_4112000_reflectance.h5")
+
+```
+
+## Create new HDF5 file framework
+
+Now, we create a new empty HDF5 file, and then populate this file with the
+groups that we need to save our essential information for this subset.
+Note that the function `h5createFile()` will not overwrite an existing
+file. Therefore, if you have already created or downloaded this file, the
+function will throw an error!
+
+```{r create-hdf5}
+# First, create a name for the new file
+nf <- paste0(wd, "NEON_hyperspectral_tutorial_example_subset.h5")
```

Comment:
Since this is the file used in the next lessons in the series, it should be named the same as the one being read into those tutorials.

Reply:
Yes, this is consistent with the other PRs. However, the other hyperspectrlal tutorial .Rmd files within this particular branch/PR have not been updated (to make the PRs unique to each tutorial). So, you might not see that this is consistent depending on how you are viewing the other files.